Month3/Project1/Project1/ArrayVectors.cpp

```diff
@@ -0,0 +1,280 @@
+	int Position;
+	int Shape;
+	float Radius;
+	float SideLeght;
+
+	for (int CurrentIndex = 0; CurrentIndex < ShapesNumber; CurrentIndex++)
+	{
+		Position = CurrentIndex + 1;
+		std::cout << "What shape are you looking for your position number " << Position << std::endl;
+		std::cout << "1. Circle" << std::endl;
+		std::cout << "2. Square" << std::endl;
+		Shape = ValidateIntInput();
+
+		while (Shape > 2)
+		{
+			std::cin.clear();
+			std::cin.ignore(100, '\n');
+			std::cout << "Sorry, You only can Select Cicle or Square, try again" << std::endl;
+			Shape = ValidateIntInput();
+		}
+
+		if (Shape == 1)
+		{
+			std::cout << "you select circle, now please enter a radius for the circle" << std::endl;
+			Radius = ValidateFloatInput();
+			ArrShapes[CurrentIndex] = new FCircle(Radius);
+		}
+		if (Shape == 2)
+		{
+			std::cout << "you select square, now please enter side lenght" << std::endl;
+			SideLeght = ValidateFloatInput();
+			ArrShapes[CurrentIndex] = new FSquare(SideLeght);
+		}
+
+	}
+	system("CLS");
+	for (int CurrentIndex = 0; CurrentIndex < ShapesNumber; CurrentIndex++)
+	{
+		std::cout << "The Area of your shape is: " << ArrShapes[CurrentIndex]->GetArea() << std::endl;
+		std::cout << "The Perimeter of your shape is: " << ArrShapes[CurrentIndex]->GetPerimeter() << std::endl;
+		std::cout << std::endl;
+	}
+	
+	delete[] ArrShapes;
+}
+
+
+void DinamicArray() 
+{
+	TDinamicArray <FShape*> ArrShapes(1);
+	std::cout << "What do you want to do" << std::endl;
+	std::cout << "1. Add a Shape" << std::endl;
+	std::cout << "2. You want to print the shapes you already have?" << std::endl;
+	int Choice = ValidateIntInput();
+	while (Choice > 2 || Choice < 1) 
+	{
+		std::cin.clear();
+		std::cin.ignore(100, '\n');
+		std::cout << "invalid Input, you only can choice 1 or 2 option, Try Again" << std::endl;
+		Choice = ValidateIntInput();
+	}
+	float Radius;
+	float SideLeght;
+	if (Choice == 1)
+	{
+		std::cout << "you select circle, now please enter a radius for the circle" << std::endl;
+		Radius = ValidateFloatInput();
+		ArrShapes.PushBack(new FCircle(Radius));
+	}
+
+	if (Choice == 2)
+	{
+		std::cout << "you select square, now please enter side lenght" << std::endl;
+		SideLeght = ValidateFloatInput();
+		ArrShapes.PushBack(new FSquare(SideLeght));
+	}
+	
+	std::cout << "You want to print the shapes you already have?" << std::endl;
+	std::cout << "1.Yes" << std::endl;
+	std::cout << "2. Not" << std::endl;
+	int Option = ValidateIntInput();
+	if (Option == 1) 
+	{
+		int Counter	= 1;
+		for (int CurrentIndex = 0; CurrentIndex <= ArrShapes.GetSize(); CurrentIndex++)
+		{
+			std::cout << "the Area of your shape number " << Counter << "is: " << ArrShapes[CurrentIndex]->GetArea() << std::endl;
+			std::cout << "The perimeter of Yout Shape number " << Counter << "is: " << ArrShapes[CurrentIndex]->GetPerimeter() << std::endl;
+			std::cout << std::endl;
+		}
+	}
+	if (Option == 2) 
+	{
+		return;
+	}
+}
```

Comment:
Similar to the Builtin and later the Static arrays, make sure to delete the contents of those arrays! The destructor of the array takes care of the actual array, but once again, what about the shapes they pointers are pointing to?

Reply:
This is very important! EVERY new should have it's own delete.